File: LeetCode/Medium/0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        boolean[] visit = new boolean[n];

        bfs(nums, visit, n);
        return visit[n-1];
    }

    public void bfs(int[] nums, boolean[] visit, int n) {
        Deque<Integer> q = new ArrayDeque<>();
        visit[0] = true;
        q.add(0);

        while (q.size() > 0) {
            int cur = q.poll();
            for (int i = 1; i <= nums[cur]; i++) {
                int nxt = cur + i;
                if (nxt >= n) {
                    continue;
                }
                if (!visit[nxt]) {
                    visit[nxt] = true;
                    q.add(nxt);
                }
            }
        }
    }
}

File: LeetCode/Medium/0055-jump-game/0055-jump-game.go
func canJump(nums []int) bool {
    n := len(nums)
    visit := make([]bool, n)
    var stack []int

    stack = append(stack, 0)
    visit[0] = true

    for len(stack) != 0 {
        top := len(stack) - 1
        cur := stack[top]
        stack = stack[:top]
        
        if (cur >= n) {
            continue
        }

        maxLength := nums[cur]

        for i := 1; i <= maxLength; i++ {
            if (cur + i >= n || visit[cur + i]) {
                continue
            }
            visit[cur + i] = true
            stack = append(stack, cur + i)
        }
    }
    return visit[n-1]
}